packages/kotti-ui/source/kotti-banner/KtBanner.vue

```diff
@@ -1,81 +1,109 @@
 <template>
 	<div :class="bannerClass">
-		<div class="collapse">
-			<div class="glyph">
-				<span v-if="icon">
+		<div class="kt-banner__collapse">
+			<div class="kt-banner__glyph">
+				<slot name="glyph">
 					<i class="yoco" v-text="icon" />
-				</span>
-				<span v-else>
-					<slot name="glyph">
-						<i class="yoco">announce</i>
-					</slot>
-				</span>
+				</slot>
 			</div>
-			<div class="message" v-text="message" />
-			<div v-if="!expandable" class="action" @click="handleClick">
-				<button
-					v-if="actionText.length"
-					class="kt-button text"
-					v-text="actionText"
-				/>
+			<div class="kt-banner__message" v-text="message" />
+			<div v-if="!expandable" @click="() => $emit('click')">
+				<KtButton v-if="actionText.length" :label="actionText" type="text" />
 			</div>
-			<div v-else class="action" @click="isExpand = !isExpand">
-				<button v-if="!isExpand" class="kt-button text" v-text="switchText" />
-				<button v-else class="kt-button text" v-text="switchCloseText" />
+			<div v-else @click="isExpand = !isExpand">
+				<KtButton v-if="!isExpand" :label="switchText" type="text" />
+				<KtButton v-else :label="switchCloseText" type="text" />
 			</div>
 		</div>
-		<div v-if="isExpand" class="expand">
+		<div v-if="isExpand" class="kt-banner__expand">
 			<slot name="expand" />
 		</div>
 	</div>
 </template>
 
-<script>
-export default {
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent, ref } from '@vue/composition-api'
+
+import { KtButton } from '../kotti-button'
+import { isYocoIcon } from '../validators'
+
+import { KottiBanner } from './types'
+
+export default defineComponent<KottiBanner.PropsInternal>({
 	name: 'KtBanner',
+	components: { KtButton },
 	props: {
-		isGrey: {
-			type: Boolean,
-			default: false,
-		},
-		message: String,
+		actionText: { type: String, default: '' },
 		icon: {
 			type: String,
-			default: '',
-		},
-		actionText: {
-			type: String,
-			default: '',
-		},
-		switchText: {
-			type: String,
-			default: 'View',
-		},
-		switchCloseText: {
-			type: String,
-			default: 'Close',
+			default: Yoco.Icon.ANNOUNCE,
+			validator: isYocoIcon,
 		},
+		isGray: { type: Boolean, default: false },
+		message: { type: String, required: true },
+		switchCloseText: { type: String, default: 'Close' },
```

Comment:
Is this heavily utilized and often kept as-is in yoda? If yes, then ideally this would be covered by kotti translations